contents/product-engineers/trunk-based-development.md

```diff
@@ -0,0 +1,122 @@
+
+### 1. Avoiding merge conflicts
+
+Merge conflicts happen when attempting to merge branches with different changes to the same areas. These are often unexpected and take time to solve. 
+
+Long-lived branches provide opportunities for merge conflicts to pile up. The short-lived branches of trunk-based development stay closer to `main` because they branch off and merge into it often. This helps avoid merge conflicts.
+
+### 2. Keeps main healthy
+
+The `main` or trunk branch is the most important branch for two reasons:
+
+1. It gets deployed or shipped to users.
+2. It is the source of truth for the codebase.
+
+By merging to `main` regularly, everyone can be confident `main` is stable and the most up-to-date code. Developers do not need to search for healthy branches. 
+
+It also enables engineers to be more confident in continuously deploying it. This helps teams ship more and maintain momentum. 
+
+### 3. Reviewable PRs
+
+Large feature branches can often have hundreds or thousands of lines of changes. Developers dread massive pull requests (PRs AKA merge requests) for a reason. Reviewing these requires a significant amount of overhead to get up to speed with the code.
+
+The smaller branches of trunk-based development create more reviewable pull requests. They require less overhead for someone to review. Merging regularly also creates opportunities for teams to collaborate. Every PR is a chance to interact with each other, get feedback earlier, and see what others are working on. 
+
+## Why and how PostHog does trunk-based development
+
+PostHog does trunk-based development. The big reason why is to **minimize work-in-progress**. This is a [core philosophy of our development process](/handbook/engineering/development-process#sizing-tasks-and-reducing-wip) at PostHog and relates to all three of the benefits of the last section.
+
+Implementing this philosophy (and trunk-based development) means:
+
+- Optimizing pull requests for quality of implementation *and* the speed of merge.
+
+- Keeping changes doable in one day, including code review and QA.
+
+- If not, split them into smaller changes until they fit into one day.
+
+- Responding to colleagues' review requests early in your day to help unblock them.
+
+- Pushing code out early. PRs are better than issues, even if they are half-done. This enables more concrete feedback on the approach.
+
+- Merging whenever. Our testing, reviewing, and build processes should be good enough that we are confident doing this.
+
+As proof of this, our median pull request age when merged is **just under 8 hours**. On an average day, we merge around 20 pull requests in our main repo.
+
+All of this helps us get the benefits trunk-based development provides. It might seem like a lot. Pushing code early and merging whenever may seem especially scary, but we have two tools that help us:
```

Comment:
Could you take a second stab at these two paras? I find them difficult to parse atm.